test/pytest/service-desktop/test_send_sms.py

```diff
@@ -32,13 +31,8 @@ def test_send_sms_too_long(harness, phone_number):
     for i in range(67*7):
         sms_text = sms_text + chr(48 + i % 10)
 
-    body = {"category": "message", "number": str(phone_number), "messageBody": sms_text}
-    ret = harness.endpoint_request("messages", "post", body)
-    assert ret["status"] == status["OK"]
-
-    sms_text = sms_text + 'a'
-    body = {"category": "message", "number": str(phone_number), "messageBody": sms_text}
-    ret = harness.endpoint_request("messages", "post", body)
-    assert ret["status"] == status["BadRequest"]
-
+    ret = AddMessage(str(phone_number), sms_text).run(harness)
 
+    sms_text = sms_text + "a"
+    with pytest.raises(TransactionError, match=r".*" + str(Status.BadRequest.value) + ".*"):
```

Comment:
why should it rise TransactionError -> BadRequest?

Reply:
It is an expected error, when we try to send a too long message.